contract_compassion/model/contracts.py

```diff
@@ -0,0 +1,646 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Cyril Sester, Emanuel Cino
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+from openerp import netsvc
+from openerp.tools.translate import _
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class recurring_contract(orm.Model):
+    _inherit = "recurring.contract"
+    _order = 'start_date desc'
+
+    def on_change_partner_id(self, cr, uid, ids, partner_id, context=None):
+        """ On partner change, we update the correspondent and
+        set the new pol_number (for gift identification). """
+        res = super(recurring_contract, self).on_change_partner_id(
+            cr, uid, ids, partner_id, context)
+        num_contracts = self.search(
+            cr, uid, [('partner_id', '=', partner_id)], context=context,
+            count=True)
+        # If contract created check state
+        if ids:
+            contract = self.browse(cr, uid, ids[0], context)
+            # If state draft correspondant_id=parent_id
+            if (contract.state == 'draft'):
+                res['value'].update({
+                    'correspondant_id': partner_id,
+                })
+        # Else correspondant_id=parent_id
+        else:
+            res['value'].update({
+                'correspondant_id': partner_id,
+            })
+        res['value'].update({
+            'num_pol_ga': num_contracts
+        })
+        return res
+
+    def on_change_next_invoice_date(
+            self, cr, uid, ids, new_invoice_date, context=None):
+        res = True
+        for contract in self.browse(cr, uid, ids, context):
+            if (contract.state not in ('draft', 'mandate')):
+                res = super(
+                    self._name, self).on_change_next_invoice_date(

```

Comment:
I'm not sure you can use self._name here, as self._name returns 'recurring.contract' and you would normally put recurring_contract (the python class name).